Lib/ufo2ft/filters/propagateAnchors.py

```diff
@@ -74,28 +74,37 @@ def _propagate_glyph_anchors(glyphSet, composite, processed):
                 mark_components.append(component)
             else:
                 base_components.append(component)
-                anchor_names |= {a.name for a in glyph.anchors}
+                anchor_names.update(a.name for a in glyph.anchors)
+
+    if mark_components and not base_components:
+        if _is_ligature_mark(composite):
+            # The composite is a mark that is composed of other marks (E.g.
+            # "circumflexcomb_tildecomb"). Promote the mark that is positioned closest
+            # to the origin to a base.
+            try:
+                component = _component_closest_to_origin(mark_components, glyphSet)
+            except Exception as e:
+                raise Exception(
+                    "Error while determining which component of composite "
+                    "'{}' is the lowest: {}".format(composite.name, str(e))
+                )
+            components = [component]
+        else:
+            components = mark_components
```

Comment:
What is the logic behind promoting _all_ mark components to bases here? This works for single component glyphs like the `macroncomb.alt` in your test but what if it's not so simple?